src/annotator/host.js

```diff
@@ -0,0 +1,102 @@
+import $ from 'jquery';
+
+// TODO - Convert this to an ES import once the `Guest` class is converted to JS.
+// @ts-expect-error
+const Guest = require('./guest');
+
+export default class Host extends Guest {
+  constructor(element, config) {
+    // Some config settings are not JSON-stringifiable (e.g. JavaScript
+    // functions) and will be omitted when the config is JSON-stringified.
+    // Add a JSON-stringifiable option for each of these so that the sidebar can
+    // at least know whether the callback functions were provided or not.
+    if (config.services?.length > 0) {
+      const service = config.services[0];
+      if (service.onLoginRequest) {
+        service.onLoginRequestProvided = true;
+      }
+      if (service.onLogoutRequest) {
+        service.onLogoutRequestProvided = true;
+      }
+      if (service.onSignupRequest) {
+        service.onSignupRequestProvided = true;
+      }
+      if (service.onProfileRequest) {
+        service.onProfileRequestProvided = true;
+      }
+      if (service.onHelpRequest) {
+        service.onHelpRequestProvided = true;
+      }
+    }
+
+    // Make a copy of all config settings except `config.sidebarAppUrl`, the app base URL,
+    // and `config.pluginClasses`
+    const configParam =
+      'config=' +
+      encodeURIComponent(
+        JSON.stringify(
+          Object.assign({}, config, {
```

Comment:
might be able to refactor this a tad more. Is there any reason to initialize sidebarAppUrl or pluginClasses to undefined? seems like that could just be omitted.

either `JSON.stringify(config)` or `JSON.stringify({...config})` if copy is needed

Reply:
The comment above this line explains that these parameters should not be included in the config passed in the sidebar's fragment URL. I'll rewrite this to make it clearer.